test/run_envoy_fuzz_coverage.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+set -e
+
+[[ -z "${SRCDIR}" ]] && SRCDIR="${PWD}"
+
+echo "Starting run_envoy_fuzz_coverage.sh..."
+echo "    PWD=$(pwd)"
+echo "    SRCDIR=${SRCDIR}"
+
+# This is the fuzz target that will be run to generate coverage data.
+if [[ $# -gt 0 ]]; then
+  FUZZ_TARGETS=$*
+else
+  # By default, this will be all fuzz targets.
+  FUZZER_TARGETS_CC=$(find . -name *_fuzz_test.cc)
+  FUZZER_TARGETS="$(for t in ${FUZZER_TARGETS_CC}; do echo "${t:2:-3}"; done)"
+  FUZZ_TARGETS=""
+  for t in ${FUZZER_TARGETS}
+  do
+    FUZZ_TARGETS+="//"$(dirname "$t")":"$(basename "$t")" "
+  done
+fi
+
+echo ${FUZZ_TARGETS}
+
+# Build all fuzz targets to run instrumented with libfuzzer in sequence.
+echo "Building fuzz targets..."
+for t in ${FUZZ_TARGETS}
+do
+  bazel build "${t}_with_libfuzzer" --config asan-fuzzer -c opt
+done
+
+# Now run each fuzz target in parallel for 60 seconds.
+pids=""
+TEMP_CORPORA=""
+echo "Running fuzz targets..."
+for t in ${FUZZ_TARGETS}
+do
+  # Get the original corpus for the fuzz target
+  ORIGINAL_CORPUS=$(bazel query "labels(srcs, ${t}_corpus)" | head -1)
+  ORIGINAL_CORPUS=${ORIGINAL_CORPUS/://}
+  ORIGINAL_CORPUS=$(dirname ${ORIGINAL_CORPUS})
+  # Create temp directory in target's corpus
+  CORPUS_DIR=$(mktemp -d -p $(pwd)/${ORIGINAL_CORPUS:2})
+  TEMP_CORPORA+="${CORPUS_DIR} "
+  # Run fuzzing process.
+  TARGET_BINARY="${t/://}"
+  bazel-bin/${TARGET_BINARY:2}_with_libfuzzer -max_total_time=60 ${CORPUS_DIR} $(pwd)${ORIGINAL_CORPUS:1} &
+  pids="$pids $!"
+done
+
+# Wait for background process to run.
+# TODO? Processes will still run in background if user ctrl-c.
+for pid in $pids; do
+  wait ${pid}
+  if [ $? -eq 0 ]; then
+    echo "SUCCESS for $pid"
+  else
+    echo "FAILED for $pid"
+  fi
+done
+
+# This will be used by llvm-cov and points to the binaries.
+OBJECTS=""
+
+# TODO can I re-use cached targets from bazel build and bazel coverage?
+echo "Running bazel coverage for each target..."
+for t in ${FUZZ_TARGETS}
```

Comment:
As above, I think it might be possible to flatten this into a single `bazel coverage` invocation.

Reply:
This one's tricky because the test_arg has to be the corresponding corpus. Unless there's a way to give test_args flattened....

Reply:
I could change this to the normal cc_test target not linked to libfuzzer, but actually assumed this would be faster to invoke if I reused the builds. maybe not?

Reply:
Changing any flags basically forces a full rebuild.